src/service/navigation.js

```diff
@@ -529,24 +530,82 @@ export class Navigation {
    * Handles click-navigation on a non-A2A, standard-protocol link.
    * @param {!Event} e
    * @param {!Element} element
-   * @param {!Location} targetLocation
-   * @param {!Location} currentLocation
+   * @param {!Location} toLocation
+   * @param {!Location} fromLocation
    * @private
    */
-  handleNavigation_(e, element, targetLocation, currentLocation) {
-    const target = getHrefMinusHash(targetLocation);
-    const current = getHrefMinusHash(currentLocation);
+  handleNavigation_(e, element, toLocation, fromLocation) {
+    const to = getHrefMinusHash(toLocation);
+    const from = getHrefMinusHash(fromLocation);
 
     // Handle same-page (hash) navigation separately.
-    if (targetLocation.hash && target == current) {
-      this.handleHashNavigation_(e, targetLocation, currentLocation);
+    if (toLocation.hash && to == from) {
+      this.handleHashNavigation_(e, toLocation, fromLocation);
     } else {
       // Otherwise, this is an other-page (external) navigation.
+      let target = (element.getAttribute('target') || '').toLowerCase();
+
       if (this.isEmbed_ || this.isInABox_) {
         // Target in the embed must be either _top or _blank (default).
-        const targetAttr = (element.getAttribute('target') || '').toLowerCase();
-        if (targetAttr != '_top' && targetAttr != '_blank') {
-          element.setAttribute('target', '_blank');
+        if (target != '_top' && target != '_blank') {
+          target = '_blank';
+          element.setAttribute('target', target);
+        }
+      }
+
+      // Safari 13: Temporarily remove viewer query params from iframe
+      // (e.g. amp_js_v, usqp) to prevent document.referrer from being reduced
+      // to eTLD+1 (e.g. ampproject.org).
+      const {win} = this.ampdoc;
+      const platform = Services.platformFor(win);
+      const viewer = Services.viewerForDoc(element);
+      if (
+        isExperimentOn(
+          this.ampdoc.win,
+          'remove-viewer-query-params-on-navigate'
+        ) &&
+        fromLocation.search &&
+        platform.isSafari() &&
+        platform.getMajorVersion() >= 13 &&
+        viewer.isProxyOrigin() &&
+        viewer.isEmbedded()
+      ) {
+        dev().info(
+          TAG,
+          'Removing iframe query string before navigation:',
+          fromLocation.search
+        );
+        const original = fromLocation.href;
+        const noQuery = `${fromLocation.origin}${fromLocation.pathname}${fromLocation.hash}`;
+        win.history.replaceState(null, '', noQuery);
+
+        const restoreQuery = () => {
+          const currentHref = win.location.href;
+          if (currentHref == noQuery) {
+            dev().info(TAG, 'Restored iframe URL with query string:', original);
+            win.history.replaceState(null, '', original);
+          } else {
+            dev().error(
+              TAG,
+              'Unexpected iframe URL change:',
+              currentHref,
```

Comment:
We can't log these into the Error tracker due to PII.

Reply:
Should be safe since it's just the page URL with/without query params.